### PR TITLE
Update eth-recovery docs

### PR DESCRIPTION
**Pull request by gastonponti**
### Description

Updates eth-recovery docs that uses a wrong flag and generates confusion

### CODE REVIEW

File: packages/docs/celo-holder-guide/eth-recovery.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ Write your recovery phrase to a file using the following commands:
 
 1. `nano recovery.txt`
 2. Paste `<word1> <word2> … <word24>`
-3. Replace the `<word>`s in brackets with the 24 words from your recovery phrase
+3. Replace the `<word>`s in brackets with the words from your recovery phrase (usually 24 words, but works for another mnemonic sizes)
```
Comment (oneeman) [inline packages/docs/celo-holder-guide/eth-recovery.md]:
```suggestion
3. Replace the `<word>`s in brackets with the words from your recovery phrase (usually 24 words, but can be 12, 15, 18, 21 or 24 words, as specified in the [BIP 39 standard](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki))
```

File: packages/docs/celo-holder-guide/eth-recovery.md [hunk 2]
```diff
@@ -46,6 +46,12 @@ This command will return you with:
 - `privateKey`: the private key associated with your address -- please record this private key on paper and not share with anyone else
 - `publicKey`: the public key associated with your address
 
+***Note***
+
+If the address was generated with another derivationPath or address index (the ethereum's default is `"m/44'/60'/0'/0/0"` which will cover almost every case) you can specify both using the flags `addressIndex` and `derivationPath`.
+
+Example for `m/44'/78'/1'/1/23` derivationPath: `celocli account:new --mnemonicPath recovery.txt --derivationPath "m/44'/78'/1'/1" --addressIndex 23 --node https://forno.celo.org`.
```
Comment (oneeman) [inline packages/docs/celo-holder-guide/eth-recovery.md]: This isn't quite right.  The command as given will use the path `m/44'/78'/1'/1/0/23`, because it defaults to `changeAddress` being 0 (can be verified using https://iancoleman.io/bip39/).  To get `m/44'/78'/1'/1/23` you could use:
 
```
celocli account:new --mnemonicPath recovery.txt --derivationPath "m/44'/78'/1'" --changeAddress 1 --addressIndex 23 --node https://forno.celo.org
```

However, this relates to a difficulty I've run into helping a user recover their funds.  They had used an address generated using a Ledger in 2017, at which point Ledger was using a different derivation path: `m/44'/60'/0'/[addressIndex]` rather than `m/44'/60'/0'/[addressIndex]` like they do now.  Unfortunately, since the changeAddress part of the path in our code isn't optional (if you omit it, it defaults to zero rather than no change address), we can't handle this path using celocli (I had to point the user to https://iancoleman.io/bip39/).  One way to possible resolve this would be to support `--changeAddress null` or something like that, though I don't know what syntax would give the best UX.  I imagine we should file this as a separate issue, but I would certainly like to see it resolved in some way.

**Review by oneeman — COMMENTED** [review]
Much needed update!  I wrote down a couple of comments.  Regarding the derivation addresses, I think it's fine to just update them to make clear the role of the change index as well, but in the future I certainly would like to see the command itself updated to support paths such as the legacy Ledger path.

File: packages/docs/celo-holder-guide/eth-recovery.md (hunk 2 above)
Comment (oneeman) [reply packages/docs/celo-holder-guide/eth-recovery.md]: Or perhaps we could add another special preset `ethLegacyLedger`.  But it would still have to involve some change to `generateKeys()` so it can omit the change portion of the path.  See https://github.com/celo-org/celo-monorepo/blob/2adcf1eeca33910387e0c4e5fde3afda8f035901/packages/sdk/utils/src/account.ts#L167-L184

Comment (gastonponti) [reply packages/docs/celo-holder-guide/eth-recovery.md]: Ok, I'll change the docs for now, and we could add an issue to fix that

File: packages/docs/celo-holder-guide/eth-recovery.md [hunk 3]
```diff
@@ -46,6 +46,12 @@ This command will return you with:
 - `privateKey`: the private key associated with your address -- please record this private key on paper and not share with anyone else
 - `publicKey`: the public key associated with your address
 
+***Note***
+
+If the address was generated with another derivationPath or address index (the ethereum's default is `"m/44'/60'/0'/0/0"` which will cover almost every case) you can specify both using the flags `addressIndex`, `changeIndex` and `derivationPath`.
```
Comment (oneeman) [inline packages/docs/celo-holder-guide/eth-recovery.md]:
```suggestion
Using the `eth` derivation path as above will work for the default Ethereum path used by nearly all Ethereum wallets (`"m/44'/60'/0'/0/0"`). If your address was generated using a different derivation path you can specify that using a combination of the flags `addressIndex`, `changeIndex` and `derivationPath`.
```

File: packages/docs/celo-holder-guide/eth-recovery.md [hunk 4]
```diff
@@ -46,6 +46,12 @@ This command will return you with:
...
+
+Example for `m/44'/78'/1'/4/23` derivationPath: `celocli account:new --mnemonicPath recovery.txt --derivationPath "m/44'/78'/1'" --changeIndex 4 --addressIndex 23 --node https://forno.celo.org`.
```
Comment (oneeman) [inline packages/docs/celo-holder-guide/eth-recovery.md]: I would rephrase it as follows (can't use a github suggestion because of the "```" being used both for suggesting and for the content).

For example, for the address corresponding to the path `m/44'/78'/1'/4/23` use: 
```
celocli account:new --mnemonicPath recovery.txt --derivationPath "m/44'/78'/1'" --changeIndex 4 --addressIndex 23 --node https://forno.celo.org
```

**Review by oneeman — COMMENTED** [review]
Suggested a couple of rephrasings, otherwise looks good.

**Review by barbaraliau — APPROVED** [review]
LGTM. Thanks for writing this up! +1 to @oneeman's comments.
